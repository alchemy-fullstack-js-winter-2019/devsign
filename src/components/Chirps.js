import React from 'react';
import PropTypes from 'prop-types';
import Chirp from './Chirp';
import { List } from '../styles/mainStyle';

function Chirps({ chirps }) {

  const chirpList = chirps.map((chirp, i) => {
    return <li key={i}><Chirp
      profileImage={chirp.user.profileImage}
      name={chirp.user.name} 
      text={chirp.text}
      handle={chirp.user.handle}
    /></li>;
  });
  return (
    <List>
      {chirpList}
    </List>
  );
}

Chirps.propTypes = {
  chirps: PropTypes.array.isRequired
};

export default Chirps;
