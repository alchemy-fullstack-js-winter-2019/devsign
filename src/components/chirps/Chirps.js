import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Chirp from './Chirp';

function Chirps({ chirps, className }) {
  const chirpItems = chirps.map(chirp => {
    return <li key={chirp._id}><Chirp chirp={chirp} /></li>;
  });

  return (
    <ul className={className}>
      {chirpItems}
    </ul>
  );
}

Chirps.propTypes = {
  chirps: PropTypes.array.isRequired,
  className: PropTypes.string.isRequired
};

export default styled(Chirps)`
  list-style-type: none;
  padding: 0;
`;