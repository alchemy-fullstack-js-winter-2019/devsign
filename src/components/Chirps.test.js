import React from 'react';
import Chirps from './Chirps';
import renderer from 'react-test-renderer';

describe('Chirps component', () => {
  it('matches a snapshot', () => {
    const chirps = [
      {
        user: { profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
          name: 'Ray',
          handle: '@stingraymond'  },
        text: 'tweet tweet tweet'
      },
      {
        user: { profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
          name: 'Kaiya',
          handle: '@ladybeard'  },
        text: 'tweet tweet tweet'
      },
      {
        user: { profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
          name: 'Randy',
          handle: '@randyroute' },
        text: 'tweet tweet tweet'
      }
    ];
    const tree = renderer.create(
      <Chirps chirps={chirps}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
