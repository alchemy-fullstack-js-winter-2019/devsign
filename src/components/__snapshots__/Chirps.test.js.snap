// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chirps component matches a snapshot 1`] = `
<ul
  className="sc-iwsKbI jzeEWW"
>
  <li>
    <article
      className="sc-jzJRlG dFsfHb"
    >
      <header
        className="sc-chPdSV iJfIXb"
      >
        <img
          alt="Ray"
          className="sc-jTzLTM fkBueS"
          src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
        />
        <h2
          className="sc-cSHVUG eRmrdv"
        >
          Ray
           
          <span
            className="sc-fjdhpX FzRiJ"
          >
            @stingraymond
          </span>
        </h2>
      </header>
      <p
        className="sc-gqjmRU sc-kAzzGY ctCWvU"
      >
        tweet tweet tweet
      </p>
    </article>
  </li>
  <li>
    <article
      className="sc-jzJRlG dFsfHb"
    >
      <header
        className="sc-chPdSV iJfIXb"
      >
        <img
          alt="Kaiya"
          className="sc-jTzLTM fkBueS"
          src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
        />
        <h2
          className="sc-cSHVUG eRmrdv"
        >
          Kaiya
           
          <span
            className="sc-fjdhpX FzRiJ"
          >
            @ladybeard
          </span>
        </h2>
      </header>
      <p
        className="sc-gqjmRU sc-kAzzGY ctCWvU"
      >
        tweet tweet tweet
      </p>
    </article>
  </li>
  <li>
    <article
      className="sc-jzJRlG dFsfHb"
    >
      <header
        className="sc-chPdSV iJfIXb"
      >
        <img
          alt="Randy"
          className="sc-jTzLTM fkBueS"
          src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
        />
        <h2
          className="sc-cSHVUG eRmrdv"
        >
          Randy
           
          <span
            className="sc-fjdhpX FzRiJ"
          >
            @randyroute
          </span>
        </h2>
      </header>
      <p
        className="sc-gqjmRU sc-kAzzGY ctCWvU"
      >
        tweet tweet tweet
      </p>
    </article>
  </li>
</ul>
`;
